repose-aggregator/tests/functional-tests/src/test/groovy/features/filters/ipuser/IpUserTest.groovy

```diff
@@ -117,4 +117,65 @@ class IpUserTest extends ReposeValveTest {
+        given: "x-forwarded-for header is set"
+        def params = properties.defaultTemplateParams
+        repose.configurationProvider.applyConfigs("common", params)
+        repose.configurationProvider.applyConfigs("features/filters/ipuser", params)
+        repose.start()
+        def headers = ['x-forwarded-for': '10.1.1.3']
+
+        when: "Request is sent through repose"
+        MessageChain mc = deproxy.makeRequest(url: reposeEndpoint, method: 'get', headers: headers)
+        def sentRequest = mc.handlings[0]
+
+        then: "Repose will send x-pp-group with the configured value"
+        mc.handlings.size() == 1
+
+        and: "Repose will x-pp-user base on x-forward-for header plus default quality"
+        sentRequest.request.headers.findAll("x-pp-user").size() == 1
+        sentRequest.request.headers.getFirstValue("x-pp-user") == '10.1.1.3;q=0.4'
+    }
+
+    def "When multi x-forwarded-for headers are included shoud take 1st valued for x-pp-user"() {
+        given: "x-forwarded-for header is set"
+        def params = properties.defaultTemplateParams
+        repose.configurationProvider.applyConfigs("common", params)
+        repose.configurationProvider.applyConfigs("features/filters/ipuser", params)
+        repose.start()
+        def headers = ['x-forwarded-for': '192.25.25.15',
+                       'X-Forwarded-For': '10.1.1.3']
+        when: "Request is sent through repose"
+        MessageChain mc = deproxy.makeRequest(url: reposeEndpoint, method: 'get', headers: headers)
+        def sentRequest = mc.handlings[0]
+
+        then: "Repose will send x-pp-group with the configured value"
+        mc.handlings.size() == 1
+
+        and: "Repose will x-pp-user base on x-forward-for header plus default quality"
+        sentRequest.request.headers.findAll("x-pp-user").size() == 1
+        sentRequest.request.headers.getFirstValue("x-pp-user") == '192.25.25.15;q=0.4'
+    }
+
+    def "When multi x-forwarded-for splitable headers are included shoud take 1st valued for x-pp-user"() {
+        given: "x-forwarded-for header is set"
+        def params = properties.defaultTemplateParams
+        repose.configurationProvider.applyConfigs("common", params)
+        repose.configurationProvider.applyConfigs("features/filters/ipuser", params)
+        repose.start()
+        def headers = ['x-forwarded-for': '192.25.25.15, 10.1.1.3']
+
+        when: "Request is sent through repose"
+        MessageChain mc = deproxy.makeRequest(url: reposeEndpoint, method: 'get', headers: headers)
+        def sentRequest = mc.handlings[0]
+
+        then: "Repose will send x-pp-group with the configured value"
+        mc.handlings.size() == 1
+
+        and: "Repose will x-pp-user base on x-forward-for header plus default quality"
+        sentRequest.request.headers.findAll("x-pp-user").size() == 1
+        sentRequest.request.headers.getFirstValue("x-pp-user") == '192.25.25.15;q=0.4'
+    }
```

Comment:
These new tests and the `def "classifying a request by its IP"()` test don't need to restart Repose in between tests do they?